lldb/source/Plugins/ExpressionParser/Swift/SwiftExpressionParser.cpp

```diff
@@ -1969,30 +1971,21 @@ SwiftExpressionParser::Parse(DiagnosticManager &diagnostic_manager,
     m_module.reset(ContextAndModule.second);
   }
 
-  if (m_swift_ast_ctx.HasErrors()) {
+  // If IRGen failed without errors, the root cause may be a fatal
+  // Clang diagnostic.
+  if (m_swift_ast_ctx.HasErrors() || m_swift_ast_ctx.HasClangImporterErrors()) {
+    diagnostic_manager.PutString(eDiagnosticSeverityRemark,
+                                 "couldn't IRGen expression.");
```

Comment:
The fact that no error is provided is pretty interesting and probably worth mentioning to the user (if only to show that we couldn't do better). How about "couldn't IRGen expression: failed without error message" or something along those lines?

Reply:
Maybe this should go below where we encourage the user to enable logging.

Reply:
That's not what's happening here. The case where there is no error message is the one below where we encourage the logging.